### DISCUSSION TITLE
Setting up wss in CapRover / Access to Let's Encrypt certs needed?

### DISCUSSION THREAD

**Discussion by tri-bit** [caprover/caprover#970] [Q&A] [answered]
I'm trying to setup a wss:// connection via node in one of my apps, I've turned on the websocket option but I think I have to create a secure server using the domain's certificate for ssl to work with websockets. Is there a way to access the Let's Encrypt cert.pem and key.pem files? 

The [ws package](https://www.npmjs.com/package/ws) code that creates the server using the cert files:

`const server = https.createServer({  
  cert: fs.readFileSync('/path/to/cert.pem'),  
  key: fs.readFileSync('/path/to/key.pem')  
});
`

**Comment by githubsaturn** [comment, marked as answer]
Your NodeJS container is behind an nginx proxy that performs SSL termination. You don't need to read the keys in your nodejs code.

**Reply by tri-bit** [reply]
I was able to get wss working without the need for the certificate keys. Thank you for all your work on this great project.
